### ISSUE TITLE
buildkit secrets not working? (help)

### ISSUE THREAD

**Issue by antonysouthworth-halter** [vito/oci-build-task#64]
I am using the following `task.yml`

```
platform: linux
image_resource:
  type: registry-image
  source:
    repository: vito/oci-build-task
params:
  CONTEXT: workdir
  BUILDKIT_SECRET_mysecret: workdir/mysecret.txt
outputs:
  - name: image
  - name: workdir
run:
  path: sh
  args:
    - -euxo
    - pipefail
    - -c
    - |
      cat >workdir/Dockerfile <<EOF
      # syntax=docker/dockerfile:1.2

      FROM alpine

      # shows secret from default secret location:
      RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret

      # shows secret from custom secret location:
      RUN --mount=type=secret,id=mysecret,dst=/foobar cat /foobar
      EOF

      echo -n "test value" >workdir/mysecret.txt

      build
```

When running with `fly execute -c task.yml --privileged`, I get the following output:

<details>
<summary>(click to expand)</summary>

```
initializing
selected worker: i-007d43aa9569b607e
INFO: found resource cache from local cache

selected worker: i-065a8ff9e1dc71e02
running sh -euxo pipefail -c cat >workdir/Dockerfile <<EOF
# syntax=docker/dockerfile:1.2

FROM alpine

# shows secret from default secret location:
RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret

# shows secret from custom secret location:
RUN --mount=type=secret,id=mysecret,dst=/foobar cat /foobar
EOF

echo -n "test value" >workdir/mysecret.txt

build

+ cat
+ echo -n 'test value'
+ build
INFO[0000] building image
#1 [internal] load build definition from Dockerfile
#1 sha256:480d890de9b69b5da8631b479d94af2b21331184cfdf12df2727992110a38743
#1 transferring dockerfile: 296B done
#1 DONE 0.0s

#2 [internal] load .dockerignore
#2 sha256:07db6c4d1fd4febdc3b0ff744acb853ba5289dd81706899585ee6ed1c4c4ecce
#2 transferring context: 2B done
#2 DONE 0.0s

#3 resolve image config for docker.io/docker/dockerfile:1.2
#3 sha256:b239a20f31d7f1e5744984df3d652780f1a82c37554dd73e1ad47c8eb05b0d69
#3 DONE 2.7s

#4 docker-image://docker.io/docker/dockerfile:1.2@sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7d313c3e2230a333fdbcc
#4 sha256:37e0c519b0431ef5446f4dd0a4588ba695f961e9b0e800cd8c7f5ba6165af727
#4 resolve docker.io/docker/dockerfile:1.2@sha256:e2a8561e419ab1ba6b2fe6cbdf49fd92b95912df1cf7d313c3e2230a333fdbcc done
#4 sha256:3cc8e449ce9f6e0752ede8f50a7334bf0c7b2d24d76da2ffae7aa6a729dd1da4 0B / 9.64MB 0.2s
#4 sha256:3cc8e449ce9f6e0752ede8f50a7334bf0c7b2d24d76da2ffae7aa6a729dd1da4 9.64MB / 9.64MB 0.3s done
#4 extracting sha256:3cc8e449ce9f6e0752ede8f50a7334bf0c7b2d24d76da2ffae7aa6a729dd1da4
#4 extracting sha256:3cc8e449ce9f6e0752ede8f50a7334bf0c7b2d24d76da2ffae7aa6a729dd1da4 0.1s done
#4 DONE 0.4s

#5 [internal] load metadata for docker.io/library/alpine:latest
#5 sha256:d4fb25f5b5c00defc20ce26f2efc4e288de8834ed5aa59dff877b495ba88fda6
#5 DONE 1.3s

#6 [1/3] FROM docker.io/library/alpine@sha256:69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f
#6 sha256:c46b598ce45cd8cd40f4705fe76968d7d0b56fd99853cf12e4fad77079938f1d
#6 resolve docker.io/library/alpine@sha256:69e70a79f2d41ab5d637de98c1e0b055206ba40a8145e7bddb55ccc04e13cf8f done
#6 sha256:540db60ca9383eac9e418f78490994d0af424aab7bf6d0e47ac8ed4e2e9bcbba 2.81MB / 2.81MB 0.3s done
#6 extracting sha256:540db60ca9383eac9e418f78490994d0af424aab7bf6d0e47ac8ed4e2e9bcbba 0.1s done
#6 DONE 0.3s

#7 [2/3] RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
#7 sha256:a132a773ff5e6ff464dda89ce2cbabaf875360f594853fedc96bbd2e9e88edf0
#7 0.133 cat: can't open '/run/secrets/mysecret': No such file or directory
#7 ERROR: executor failed running [/bin/sh -c cat /run/secrets/mysecret]: exit code: 1
------
 > [2/3] RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret:
------
Dockerfile:6
--------------------
   4 |
   5 |     # shows secret from default secret location:
   6 | >>> RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
   7 |
   8 |     # shows secret from custom secret location:
--------------------
error: failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c cat /run/secrets/mysecret]: exit code: 1
FATA[0007] failed to build: build: exit status 1
FATA[0007] failed to run task: exit status 1
failed
```

</details>

You can see, apparently the secrets are not being mounted inside?

```
#7 [2/3] RUN --mount=type=secret,id=mysecret cat /run/secrets/mysecret
#7 sha256:a132a773ff5e6ff464dda89ce2cbabaf875360f594853fedc96bbd2e9e88edf0
#7 0.133 cat: can't open '/run/secrets/mysecret': No such file or directory
```

@multimac is there any extra configuration, perhaps inside of the Concourse deployment itself, that is required for this feature to work? Also, could you post the `task.yml` you used to test in #36?

**Comment by antonysouthworth-halter** [comment]
Ahh, I think I just got it; specifying `tag: master` for the image resource seems to fix the issue.

**Comment by antonysouthworth-halter** [comment]
should be fixed by #63 (i.e. release 0.8.0)
